### PR TITLE
JAMES-1670 ToRecipientFolder mailet should use init parameters

### PR DESCRIPTION
**Pull request by Ahmet-Kaplan**
I fixed the mailet, created a unit test (taken mostly from LocalDeliveryTest) and added some info about ToRecipientFolder mailet to dev-provided-mailets docs.
Hopefully I don't make any mistake :)

### CODE REVIEW

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/RecipientFolderTest.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
+/**
+ * **************************************************************
```
Comment (mbaechler) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/RecipientFolderTest.java]: I think you have formating issue on the license header

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/RecipientFolderTest.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    private MailboxManager mailboxManager;
+    private SieveRepository sieveRepository;
+    private ToRecipientFolder recipientFolder;
+    private FakeMailetConfig mailetConfig;
+
+    private String folder = null;
+
+    private String consume = null;
+
+    private void setFolder(String folder) {
```
Comment (mbaechler) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/RecipientFolderTest.java]: why do you need this setter ?

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/RecipientFolderTest.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        Mail mail = createMail();
+        recipientFolder.service(mail);
+
+        Assert.assertEquals(Boolean.valueOf(consume), mail.getState().equals(Mail.GHOST));
+    }
+
+    @Test
+    public void mailShouldBeWellReceivedByDefaultToUserWhenvirtualHostingIsTurnedOn() throws Exception {
+
+        when(usersRepository.supportVirtualHosting()).thenAnswer(new Answer<Boolean>() {
```
Comment (mbaechler) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/RecipientFolderTest.java]: thenReturn is shorter

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/RecipientFolderTest.java [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+            @Override
+            public Boolean answer(InvocationOnMock invocationOnMock) throws Throwable {
+                return true;
+            }
+        });
+        recipientTest();
+    }
+
+    private void recipientTest() throws Exception {
```
Comment (mbaechler) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/RecipientFolderTest.java]: we usually inline the test code in the test method

**Comment by mbaechler** [comment]
Thank you for your contribution, it looks good. There are only some minor style issues.

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/RecipientFolderTest.java (hunk 2 above)
Comment (Ahmet-Kaplan) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/RecipientFolderTest.java]: Those are not so necessary but I added if we may add another test which can use the parameters separately.

Comment (mbaechler) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/RecipientFolderTest.java]: We don't need setter for properties of the current object, we can access it directly

**Pull request closed without merging**